modules/gapi/src/api/gproto.cpp

```diff
@@ -212,13 +213,19 @@ void cv::validate_input_arg(const GRunArg& arg)
     case GRunArg::index_of<cv::UMat>():
     {
         const auto desc = cv::descr_of(util::get<cv::UMat>(arg));
-        GAPI_Assert(desc.size.height != 0 && desc.size.width != 0 && "incorrect dimensions of cv::UMat!"); break;
+        GAPI_Assert(validate_input_meta(desc) &&
+                    "incorrect description of cv::UMat! "
+                    "Dimensions, channel and depth must be positive");
```

Comment:
I believe the check itself and the error message here mix levels of abstractions.

Looking at this code, you don't know what is actually checked by this function, but the assert text complains about some specific fields.

I am not sure if it needs to be an assert at all. If you want provide details, provide it in the exception text you throw from the `validate_input_meta()` function (so it becomes `void` not `bool`).

Reply:
It make sense, and  "incorrect description of cv::*Mat" is enough

about exception i think it is a bad idea:
1) void function with the only reason in throwing  exception looks ugly and unsafe and inconvenient in usage and considered as bad practice
2) GAPI_Assert here means that this place is performance critical in Release build, so exception is not good here

Reply:
It is not performance critical, in the performance-critical code we use `DbgAssert`.